### ISSUE TITLE
After `Alt+Tab` out of window, `Alt` is marked "down" including after Dear ImGui regains focus

### ISSUE THREAD

**Issue by azmr** [ocornut/imgui#3532]
```
Dear ImGui 1.79 (17900)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1916
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx9
io.ConfigFlags: 0x00000000
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigWindowsMemoryCompactTimer = 60.0f
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,762.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

**Version/Branch of Dear ImGui:**

Version: 1.79
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx9.cpp + imgui_impl_win32.cpp
Compiler: cl, clang-cl
Operating System: Win7 x64

**My Issue/Question:**

When alt-tabbing out of the (win32) window with Dear ImGui in, `alt` is still marked as down after it is released.
It remains marked down when clicking or alt-tabbing back into the window.
After pressing and releasing `alt` again, it unmarks it as down.

If you hold `ctrl` or `shift` while alt-tabbing, those also get locked down.

`Key Mod` shows the correct `alt`/`ctrl`/`shift` status compared to the physical key position regardless of whether the window is focused, even when they are locked "down".

Expected behaviour: the `down` status of modifiers matches the physical key position (as accurately shown by `Key Mod`) either always or when the (win32) window is in focus.

**Screenshots/Video**
![imgui_alt_error](https://user-images.githubusercontent.com/5575948/96117212-a1ce1480-0ee1-11eb-8b0a-7bb0aeca837c.gif)

**Standalone, minimal, complete and verifiable example:**
It can be seen in the demo window from the examples. 
I used `example_win32_directx9` with `IMGUI_IMPL_WIN32_DISABLE_GAMEPAD` defined in `build_win32.bat`

**Comment by ocornut** [comment]
It's not easy to fix, 
- WM_KEYUP is not sent to unfocused windows
- arbitrary clearly all inputs while unfocused is a typical workaround but not a generally correct/valid solution

So the first question that comes to mind, is exactly which issues are you trying to fix? aka why the ALT (or other modifiers) staying up bothers you?

**Comment by azmr** [comment]
Wow, quick response! Thanks Omar :)

> which issues are you trying to fix?

This hasn't caused me any specific issues currently. I stumbled across it while debugging a different input issue (that's almost certainly in my side of the code) and thought I'd be a good citizen and mention it :grin: . So don't consider it high priority on my account.

Regarding a fix:
- Only modifiers seem to get stuck (I tried to do the same with e.g. `space` and letter keys and couldn't get them to "stick")
- You have the correct values for the modifier positions with `io.KeyAlt`, `io.KeyCtrl` etc
- Could you overwrite the `io.KeysDown` status for the modifiers using those values at an appropriate time? e.g.
   - Always (change the standard process for determining `io.KeysDown` for just the modifiers)
   - When the window isn't focused
   - When the window regains focus do a 1-off check that they match

**Comment by ocornut** [comment]
I pushed the fix suggested in #3961 : 1491d2c9168a1982ea88dd638509e64ec990daf7

As it happens, in docking + multi-viewports enabled it does make held keys flick off for one frame, but this issue has also been present in the GLFW backend and no-one noticed that before, so my conclusion is that it's a much less problematic issue than the one we are fixing now. We can improve on the multi-viewports code later.

**Comment by ocornut** [comment]
Fixed again with #4388
